### PR TITLE
feat: file list refactor

### PR DESCRIPTION
**Pull request by DiegoFigueroaSevillano**
#### Resolved task/US: ([Implement File Navigation UI]())
* * * *

## What's in this pull request?

This pull request introduces a new feature that organizes and displays project files in a hierarchical structure. The key additions include:

1. **File Organization Logic**:
   - Added a TypeScript script to convert file paths into a tree structure.
   - Differentiates between directories and files dynamically.

2. **UI Components**:
   - **FileOption**: Represents individual files as buttons with a visual indicator for selection.
   - **DirectoryOption**: Implements expandable accordions for directories, displaying nested content hierarchically.

3. **Dynamic Styling**:
   - Directory icons (`FolderIcon`) change color when expanded: white when closed and yellow (`#FFC107`) when opened.
   - Unified background color (`#2A2F40`) across all components for a cohesive look.

4. **Icon Integration**:
   - `FolderIcon` for directories and `InsertDriveFileIcon` for files are displayed to differentiate file types clearly.

5. **Enhanced User Experience**:
   - Smooth transitions and responsive design for nested directories and file selections.

### Dependencies
- **Material-UI**: Utilized for components such as `Accordion`, `Button`, and icons like `FolderIcon` and `InsertDriveFileIcon`.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have checked my code and corrected any misspellings.

### CODE REVIEW

File: src/pages/FileCoverage/FileCoverage.tsx [hunk 1]
```diff
@@ -63,33 +64,36 @@ const FileCoverage: React.FC = () => {
   if (error || errorFile) return <ErrorAdvice />;
 
   return (
-    <Container component={"section"}>
-      {selectedFilePath && filecoverageData ? (
-        <Box>
-          <Typography variant="subtitle1">{selectedFilePath}</Typography>
+    <Box display={"flex"}>
+      <FileMenuSideBar proyectFiles={[]} basePath="" />
```
Comment (fabian-romero-jalafund) [inline src/pages/FileCoverage/FileCoverage.tsx]: You have a typo here, it is `projectFiles`

File: src/components/FilesSideBar/FileMenuSideBar.tsx [hunk 2]
```diff
@@ -1,42 +1,117 @@
...
+  function onFileSelected(value: string, path: string): void {
+    setFileSelected(value);
+    setSelectedFilePath(path);
+  }
+
+  const handleDrawerToggle = () => {
+    setMobileOpen(!mobileOpen);
+  };
+
+  const DrawerContent = (
```
Comment (fabian-romero-jalafund) [inline src/components/FilesSideBar/FileMenuSideBar.tsx:49]: Please move this component to another file, you have two components in the same file

File: src/components/FilesSideBar/FileMenuSideBar.test.tsx [hunk 3]
```diff
@@ -1,44 +1,256 @@
...
+    const filePaths = [
+      "projects/base/projectFiles/dir1/file1.java",
+      "projects/base/projectFiles/dir1/subdir1/file2.java",
+      "projects/base/projectFiles/dir2/file3.java",
+    ];
+    const basePath = "projects/base/projectFiles/";
+
+    const result = organizeFiles(filePaths, basePath);
+
+    // Expected structure
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuSideBar.test.tsx]: Unnecessary comment, delete it

**Review by fabian-romero-jalafund — APPROVED** [review]

File: src/components/FilesSideBar/FileMenuOption/DirectoryOption.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  root = true,
+}) => {
+  const [expanded, setExpanded] = useState(false);
+
+  return (
+    <Box>
+      <Accordion
+        disableGutters
+        sx={{
+          backgroundColor: "#2A2F40",
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuOption/DirectoryOption.tsx]: For the bgcolor, use the colors in styleConstant, like:
<Typography color={COLORS.GREEN} >HOLAAAA</Typography>

File: src/components/FilesSideBar/FileMenuOption/DirectoryOption.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          color: "white",
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuOption/DirectoryOption.tsx]: For the color, use the colors in styleCOnstant, like

File: src/components/FilesSideBar/FileMenuOption/FileOption.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+const FileOption: React.FC<FileOptionProps> = ({
+  node,
+  setSelected,
+  isSelected,
+}) => {
+  return (
+    <Box
+      bgcolor={isSelected ? "#12141D" : "#2A2F40"}
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuOption/FileOption.tsx:20]: For the bgcolor, use the colors in styleConstant

File: src/components/FilesSideBar/FileMenuOption/FileOption.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      padding={"5px"}
+      display="flex"
+      alignItems="center"
+    >
+      <Button
+        onClick={() => setSelected()}
+        fullWidth
+        sx={{
+          justifyContent: "flex-start",
+          color: isSelected ? "white" : "gray",
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuOption/FileOption.tsx]: For the color, use the colors in styleCOnstant

File: src/components/FilesSideBar/FileMenuSideBar.tsx [hunk 8]
```diff
@@ -1,42 +1,120 @@
...
+        onClose={handleDrawerToggle}
+        ModalProps={{ keepMounted: true }}
+        sx={{
+          width: DRAWER_WIDTH,
+          flexShrink: 0,
+          "& .MuiDrawer-paper": {
+            width: DRAWER_WIDTH,
+            boxSizing: "border-box",
+            position: isLargeScreen ? "relative" : "fixed",
+            backgroundColor: "#1D212F",
```
Comment (samuel-escalera-jalafund) [inline src/components/FilesSideBar/FileMenuSideBar.tsx:110]: For the color, use the colors in styleConstant

File: src/components/Header/Header.tsx [hunk 9]
```diff
@@ -31,7 +31,7 @@ const Header: React.FC = () => {
     {
       path: "file",
       title: "File Coverage",
-      icon: <FindInPageIcon sx={{ color: "white" }} />,
+      icon: <SourceIcon sx={{ color: "white" }} />,
```
Comment (samuel-escalera-jalafund) [inline src/components/Header/Header.tsx:34]: For the color, use the colors in styleConstant

**Review by axel-ayala-jalafund — APPROVED** [review]

**Review by samuel-escalera-jalafund — APPROVED** [review]
LGTM

**Pull request merged**